### PR TITLE
Add Minify and Flatten for In-cluster deployment with Minikube

### PR DESCRIPTION
**Pull request by Alero-Awani**
**Description**

This PR introduces a central function `ProcessConfig` for processing kubeconfig files, encapsulating the functionality for loading, validating, and writing kubeconfig files, whether from a file location or a []byte input.

The function has been integrated into the `DetectKubeConfig`, addressing [this issue](https://github.com/meshery/meshery/issues/11546) which specifically occurs when running the command ` mesheryctl system start -p Kubernetes `.

Fixes #11546

**[Signed commits](../CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by leecalcote** [comment]
Thanks for tackling this, @Alero-Awani

**Review by MUzairS15 — APPROVED** [review]
LGTM

**Comment by MUzairS15** [comment]
@Alero-Awani Can you fix these lint failures, then we are good to go.

File: utils/kubernetes/client.go [hunk 1]
```diff
@@ -32,40 +33,71 @@ func DetectKubeConfig(configfile []byte) (config *rest.Config, err error) {
...
+	if err := clientcmd.Validate(*config); err != nil {
+		return nil, nil, ErrValidateConfig(err)
 	}
 
-	err = clientcmd.Validate(*cfg)
+	// Convert the config to []byte
+	configBytes, err := clientcmd.Write(*config)
 	if err != nil {
-		return nil, ErrValidateConfig(err)
+		return nil, nil, err
```
Comment (MUzairS15) [inline utils/kubernetes/client.go:92]: Please reuse/define a new meshkit error.

File: utils/kubernetes/client.go [hunk 2]
```diff
@@ -32,40 +33,71 @@ func DetectKubeConfig(configfile []byte) (config *rest.Config, err error) {
...
+	case string:
+		config, err = clientcmd.LoadFromFile(v)
+	case []byte:
+		config, err = clientcmd.Load(v)
+	default:
+		return nil, nil, ErrLoadConfig(err)
+	}
 	if err != nil {
-		return nil, ErrLoadConfig(err)
+		return nil, nil, err
```
Comment (MUzairS15) [inline utils/kubernetes/client.go:73]: Please reuse/define a new meshkit error.

**Review by MUzairS15 — CHANGES_REQUESTED** [review]

File: utils/kubernetes/client.go (hunk 2 above)
Comment (Alero-Awani) [reply utils/kubernetes/client.go:73]: Alright Noted! Thank you for the review 😃

File: utils/kubernetes/client.go [hunk 3]
```diff
@@ -32,40 +33,71 @@ func DetectKubeConfig(configfile []byte) (config *rest.Config, err error) {
...
 
 	// Look for kubeconfig at the default path
 	path := filepath.Join(utils.GetHome(), ".kube", "config")
-	if config, err = clientcmd.BuildConfigFromFlags("", path); err == nil {
+	_, cfgFile, err := ProcessConfig(path, "")
+	if err != nil{
+		return nil, err
+	}
+	if config, err = clientcmd.RESTConfigFromKubeConfig(cfgFile); err == nil {
 		return config, err
```
Comment (MUzairS15) [inline utils/kubernetes/client.go:52]:
```suggestion
		return config, err
```

While this is not immediately due to your changes, while you are here you can improve the logic to ensure meshkit error is returned.

**Review by MUzairS15 — APPROVED** [review]
Thanks @Alero-Awani

**Pull request merged**